src/pyenphase/envoy.py

```diff
@@ -88,22 +97,53 @@ def __init__(
         host: str,
         client: httpx.AsyncClient | None = None,
         timeout: float | httpx.Timeout | None = None,
+        max_request_delay: int | None = None,
     ) -> None:
         """Initialize the Envoy class."""
         # We use our own httpx client session so we can disable SSL verification (Envoys use self-signed SSL certs)
         self._timeout = timeout or LOCAL_TIMEOUT
+        self._max_request_delay = max_request_delay or MAX_REQUEST_DELAY
         self._client = client or httpx.AsyncClient(
             verify=NO_VERIFY_SSL_CONTEXT
         )  # nosec
         self.auth: EnvoyAuth | None = None
         self._host = host
-        self._firmware = EnvoyFirmware(self._client, self._host)
+        self._firmware = EnvoyFirmware(
+            self._client, self._host, self._timeout, self._max_request_delay
+        )
         self._supported_features: SupportedFeatures | None = None
         self._updaters: list[EnvoyUpdater] = []
         self._endpoint_cache: dict[str, httpx.Response] = {}
         self.data: EnvoyData | None = None
         self._common_properties: CommonProperties = CommonProperties()
 
+        if not TYPE_CHECKING:
```

Comment:
using `TYPE_CHECKING` like this is quite unexpected. 

Can you refactor this to keep the wrappers and patch them out in the test instead?

Reply:
Yes ain't that ugly, even I could see that. 

I've re-added the decorators, they don't need any patch in test. The setback now is that the max delay has to be hard-coded, can't use the class parameter. If you know a way for that it would be neat. For now left the parameter in place.

I've also added a max repeat attempts (default 4) to solve the long probe issue, retry now stops after 4 attempts or 50 sec, whichever comes first. For setup that's ~10-20 sec, for probe still 10-20 sec for each probe but far better as before.

Also removed the test for anyio.EndOfStream error, that is really a HA issue (which I dodn't understand) that needs an except in coordinator, but that's on the HA side of life.

And removed the RetryError as well as the retry is set to signal the original error.